test/functional/legacy/autocmd_spec.lua

```diff
@@ -40,3 +41,60 @@ it('no E440 in quickfix window when autocommand invalidates undo', function()
   feed('G')
   eq('', api.nvim_get_vvar('errmsg'))
 end)
+
+-- oldtest: Test_WinScrolled_Resized_eiw()
+it('WinScrolled and WinResized events can be ignored in a window', function()
+  local screen = Screen.new()
+  write_file(
+    'Xtest_winscrolled_mouse',
+    [[
+    call setline(1, ['foo']->repeat(32))
+    set eventignorewin=WinScrolled,WinResized
+    split
+    let [g:afile,g:resized,g:scrolled] = ['none',0,0]
+    au WinScrolled * let [g:afile,g:scrolled] = [expand('<afile>'),g:scrolled+1]
+    au WinResized * let [g:afile,g:resized] = [expand('<afile>'),g:resized+1]
+    ]]
+  )
+  finally(function()
+    os.remove('Xtest_winscrolled_mouse')
+  end)
+  command('so Xtest_winscrolled_mouse')
```

Comment:
Since you don't need to edit the file, you can just use `exec()` without writing the script to a file:
```suggestion
  n.exec([[
    call setline(1, ['foo']->repeat(32))
    set eventignorewin=WinScrolled,WinResized
    split
    let [g:afile,g:resized,g:scrolled] = ['none',0,0]
    au WinScrolled * let [g:afile,g:scrolled] = [expand('<afile>'),g:scrolled+1]
    au WinResized * let [g:afile,g:resized] = [expand('<afile>'),g:resized+1]
  ]])
```